### PR TITLE
Fix adding multiple bundles

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #1311.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- add a bundle
- add it again, it should finish adding, with no errors in your console and the app circles should not appear on top of each other

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7342/

**Comment by hatched** [comment]
In the CLI deploying the same bundle again upgrades the existing applications....Maybe we want to mirror that functionality in the GUI instead of adding everything again?

**Comment by huwshimi** [comment]
@hatched that's fine, if we want to do an upgrade I'll close this and open a new issue. How does the CLI know if a bundle has been deployed before? I don't think we have a way to know that in the GUI.

**Comment by frankban** [comment]
There is no way to know that in the CLI as well. There is no concept of bundle internally in Juju.
The logic in the CLI is like:
- try to deploy an app;
- if the app is already there, with that name
  - is the bundle charm is compatible with the charm of the already deployed app?
    - if not, error
    - if so, upgrade the charm if required and continue

Later, the CLI check if the number of units declared in the bundle are already deployed for all applications. Be aware that the logic in the CLI is not straightforward, and will have to be basically duped in the GUI.

**Comment by huwshimi** [comment]
So, I'll close this and open a new issue about doing the upgrade? I do wonder if the GUI case is a bit different to the CLI as we have uncommitted state, but happy either way.

**Comment by huwshimi** [comment]
The other option would be to land this so we fix the breakage and in the future think about adding an "add as new or upgrade existing bundle" confirmation.

**Comment by frankban** [comment]
I agree with you Huw. The proper fix can be done separately. In the meanwhile, if console errors are fixed by this branch, I'd say let's land it. Code looks good.

File: jujugui/static/gui/src/app/utils/bundle-importer.js [hunk 1]
```diff
@@ -681,7 +681,10 @@ YUI.add('bundle-importer', function(Y) {
         endpoints[index][0] = record[ep[0].replace(/^\$/, '')].get('id');
       }, this);
 
-      const relationId = 'pending-' + record.args[0] + record.args[1];
+      // Use the updated app names here to avoid conflicts if apps with the
+      // same name already exist.
```
Comment (hatched) [inline jujugui/static/gui/src/app/utils/bundle-importer.js]: I don't follow this comment....updated app names from where? Maybe expand on this comment a bit?

**Comment by hatched** [comment]
QA OK, I was able to deploy the elastic search bundle, then add it again to the canvas and deploy again.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7369/

**Comment by huwshimi** [comment]
Thanks @frankban and @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**